app/jobs/regular/user_email.rb

```diff
@@ -153,7 +153,18 @@ def message_for_email(user, post, type, notification, args = nil)
       # Make sure that mailer exists
       raise Discourse::InvalidParameters.new("type=#{type}") unless UserNotifications.respond_to?(type)
 
-      email_args[:email_token] = email_token if email_token.present?
+      if email_token.present?
+        email_args[:email_token] = email_token
+
+        if type.to_s == "confirm_new_email"
+          change_req = EmailChangeRequest.find_by(new_email_token: EmailToken.find_by(token: email_token))
```

Comment:
Are we able to do this in a single query by joining on the `email_token` table?

Reply:
Out of curiosity, why do we need to validate that the `email_token` exists in the database for this code path but not the others?